pybamm/models/submodels/base_submodel.py

```diff
@@ -135,6 +135,22 @@ def domain(self, domain):
     def domain_Domain(self):
         return self._domain, self._Domain
 
+    def get_parameter_info(self, by_submodel=False):
+        """
+        Extracts the parameter information and returns it as a dictionary.
+        To get a list of all parameter-like objects without extra information,
+        use :py:attr:`model.parameters`.
+
+        Returns
+        -------
+        NotImplementedError:
+            This method is not available for direct use on submodels. It is recommended to use on the full model.
+        """
+        raise NotImplementedError(
+            "Cannot use get_parameter_info (or) print_parameter_info directly on a submodel. "
+            "Please use it on the full model."
+        )
```

Comment:
This error can be validated via `self.assertRaisesRegex` with a `with` context manager clause in a method `test_parameter_info_error()` in `test_base_submodel.py`. A phrase from this error message can be used.